igvm/vm.py

```diff
@@ -456,7 +456,7 @@ def build(self, run_puppet=True, debug_puppet=False, postboot=None):
             self.prepare_vm()
 
             if run_puppet:
-                self.run_puppet(clear_cert=True, debug=debug_puppet)
+                self.run_puppet(clear_cert=rebuild, debug=debug_puppet)
```

Comment:
If the new solution is to clean Puppet cert when a VM gets deleted, why bother clearing it on build at all? Maybe all you need to do is to call cert clearning from tests and force users to delete VMs correctly during normal operation?

Reply:
It was always the concept to clean on deletion. The old behaviour to always clean on build was what gave us this issue initially. The only situation during which we want to clean certs is when we rebuild.

Reply:
@eviscares wrote:

> It was always the concept to clean on deletion.

The diff says there was no clearing on deletion before https://github.com/innogames/igvm/pull/137/files#diff-60a07663873f7ef87bb0c4c69ede494fR561

Reply:
Yes, and that is what my proposed change is about, to clean on deletion, instead of on build, because it makes more sense that way.

Reply:
Your change adds conditional cleaning on build and on unconditional on deletion. Maybe we could get away without the former one at all.